sys/riscv/riscv/trap.c

```diff
@@ -123,9 +138,40 @@ cpu_fetch_syscall_args(struct thread *td)
 		sa->callp = &p->p_sysent->sv_table[sa->code];
 
 	sa->narg = sa->callp->sy_narg;
-	memcpy(sa->args, ap, nap * sizeof(syscallarg_t));
-	if (sa->narg > nap)
-		panic("TODO: Could we have more then %d args?", NARGREG);
+#if __has_feature(capabilities)
+	if (stack_args != NULL) {
+		uint64_t intval;
```

Comment:
Strictly speaking we should probably use `register_t`/`fueword` rather than hard-code that it's 64-bit.

Reply:
Done.